src/networking.c

```
@@ -3602,11 +3698,130 @@ static int clientMatchesFilter(client *client, clientFilter client_filter) {
+            break;
+        case 'R': /* Invalid Client flag */
+            if (!c->flag.tracking_broken_redir) return 0;
+            break;
+        case 'B': /* Tracking Bcast flag */
+            if (!c->flag.tracking_bcast) return 0;
+            break;
+        case 'd': /* Dirty CAS flag */
+            if (!c->flag.dirty_cas) return 0;
+            break;
+        case 'c': /* Close after reply flag */
+            if (!c->flag.close_after_reply) return 0;
+            break;
+        case 'u': /* Unblocked flag */
+            if (!c->flag.unblocked) return 0;
+            break;
+        case 'A': /* Close ASAP flag */
+            if (!c->flag.close_asap) return 0;
+            break;
+        case 'U': /* Unix socket flag */
+            if (!c->flag.unix_socket) return 0;
+            break;
+        case 'r': /* Readonly flag */
+            if (!c->flag.readonly) return 0;
+            break;
+        case 'e': /* No evict flag */
+            if (!c->flag.no_evict) return 0;
+            break;
+        case 'T': /* No touch flag */
+            if (!c->flag.no_touch) return 0;
+            break;
+        case 'I': /* Import source flag */
+            if (!c->flag.import_source) return 0;
+            break;
+        case 'N': /* Check for no flags */
+            if (!c->flag.replica && !c->flag.primary && !c->flag.pubsub &&
+                !c->flag.multi && !c->flag.blocked && !c->flag.tracking &&
+                !c->flag.tracking_broken_redir && !c->flag.tracking_bcast &&
+                !c->flag.dirty_cas && !c->flag.close_after_reply &&
+                !c->flag.unblocked && !c->flag.close_asap &&
+                !c->flag.unix_socket && !c->flag.readonly &&
+                !c->flag.no_evict && !c->flag.no_touch &&
+                !c->flag.import_source) {
+                return 1; /* Matches 'N' */
+            }
+            break;
+        default:
+            /* Invalid flag, return false */
```

It's worth considering whether we should reject the command with an error if invalid flags are used. Otherwise some client may try to use flags from a future version and they will be silently ignored on this version.

---

Addressed this. Now when we parse the filter, we also check whether the flag is valid or not.